packages/lyra/src/prefix-tree/trie.ts

```diff
@@ -63,16 +63,35 @@ export class Trie {
       }
     }
 
+    // avoid the recursive call if the word is the same as the term
+    const [word, docIDs] = node.getWord();
+
+    if (exact && word === term) {
+      output[word] = new Set();
+
+      handleOutput(word, docIDs, output);
+
+      return output;
```

Comment:
This looks problematic: early return here means that no prefix or fuzzy search is performed. Note that, since there is no check for the node being a terminal one, this happens even when there is no exact match.

For example, say you have only the single word `star` in the whole index. Imagine searching for `sta` allowing for prefix match. You expect to find the documents associated to `star` due to prefix match, instead the logic finds `sta` (without checking that is not a terminal node), and returns no result because there is no document associated to it.

Reply:
Thank you Luca for checking these things for me, I will try to find a final solution to the problem by following your great suggestions!